Framework/API/inc/MantidAPI/SpectrumInfoItem.h

```diff
@@ -0,0 +1,130 @@
+#ifndef MANTID_API_SPECTRUMINFOITEM_H_
+#define MANTID_API_SPECTRUMINFOITEM_H_
+
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidKernel/V3D.h"
+#include "MantidTypes/SpectrumDefinition.h"
+
+using Mantid::API::SpectrumInfo;
+using Mantid::Kernel::V3D;
+using Mantid::SpectrumDefinition;
+
+namespace Mantid {
+namespace API {
+
+/** SpectrumInfoItem
+
+SpectrumInfoItem is only created by SpectrumInfoIterator and allows users of
+the SpectrumInfoIterator object access to data from SpectrumInfo. The available
+methods include:
+- isMonitor()
+- isMaksed()
+- twoTheta()
+- signedTwoTheta()
+- l2()
+- hasUniqueDetector()
+- spectrumDefinition()
+- position()
+
+@author Bhuvan Bezawada, STFC
+@date 2018
+
+Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+
+class MANTID_API_DLL SpectrumInfoItem {
+
+public:
+  // Provide copy and move constructors
+  SpectrumInfoItem(const SpectrumInfoItem &other) = default;
+  SpectrumInfoItem &operator=(const SpectrumInfoItem &rhs) = default;
+  SpectrumInfoItem(SpectrumInfoItem &&other) = default;
+  SpectrumInfoItem &operator=(SpectrumInfoItem &&rhs) = default;
```

Comment:
Hi Simon, I'm currently trying to make the copy constructor private and default as you suggest but I'm running into some errors. I think Boost Python uses the public copy constructor somewhere and that is preventing it from compiling when it is made private.

Reply:
Check if it is your own code here that is to blame: https://github.com/mantidproject/mantid/pull/23264/files/76cc64dff4d6eae90d8274b0c6160c68fd1984c7..ff8fbbdd23b734ea001da901fa2c6bce81942832#diff-b3c3494eb5ff5081627fc195f7c35cd1R75

Reply:
Yeah I checked that - I also fixed that error by making `SpectrumInfoPythonIterator` a friend class...
This error seems to come from the file value_holder.hpp.

Reply:
Weird. To me this looks as if it should not be making a copy: https://github.com/mantidproject/mantid/pull/23264/files/76cc64dff4d6eae90d8274b0c6160c68fd1984c7..ff8fbbdd23b734ea001da901fa2c6bce81942832#diff-916598d30cb06156132f820ec939321dR25. Can you find someone more familar with the Python exporting?

Reply:
So last week I was having trouble getting the iterator to work with `DetectorInfo` and Martyn figured out that there was a bug with Boost. He gave me this workaround which I based the current solution around:
https://github.com/martyngigg/sandbox/tree/master/cpp/boost/python/iterator
He also makes the copy and move constructors public...

Reply:
Ok. So my last idea to try would be to figure out what boost class does the copy and add that as a friend as well?

Otherwise I fear we might be stuck with it, and need to make sure to add good documentation highlighing that `Item` should not be stored anywhere.

Reply:
@martyngigg Hi Martyn, just wondering if you had a minute to look at this?

Reply:
Hi Martyn. I have just tried changing the `return_value_policy` and I still get the error. Maybe it is just best to leave the copy/move constructors as they are and highlight this in the documentation as Simon suggests.

Reply:
Ah yes it is needed in another place.

I've managed to make a change to my toy example that does not require a copy: https://github.com/martyngigg/sandbox/commit/f591d36f5995aa9ca22bd308e2bf5edd97d4c26e. It also removes a part that annoyed my about the implementation where we needed to store the current item in `ContainerPythonIterator`. This now just stores a flag.

Would this work here?

Reply:
Yeah I think it works now! The tests all seem to pass. Thank you for your help on this! :)